### PR TITLE
docs: use "distributed training" to mean any kind of multi-GPU training

### PR DESCRIPTION
**Pull request by neilconway**
## Description
We previously used "parallel training" to mean "multi-GPU, single-machine
training", and "distributed training" to mean "multi-GPU, multi-machine
training". However, this isn't a useful distinction for users; also, other
packages in this space (e.g., Horovod) refer to themselves as providing
"distributed training" -- so just use that term for all kinds of multi-GPU
training.

Make other copy-editing improvements to the dtrain docs.

## Test Plan
Proof-read.

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/topic-guides/optimizing-distributed-training.txt [hunk 1]
```diff
@@ -52,8 +56,8 @@ where N is the total number of GPUs on an agent machine. In this
 configuration, trials will train using all the resources on a single machine.
 
 
-Distributed Multi-Machine Parallelism
--------------------------------------
+Multi-Machine Training
+----------------------
 
 Multi-machine parallelism offers the ability to further parallelize training across more GPUs.
 For distributed multi-machine training, in the experiment configuration, add:
```
Comment (aaron276h) [inline docs/topic-guides/optimizing-distributed-training.txt:63]: probably remove distributed

File: docs/how-to/distributed-training.txt [hunk 2]
```diff
@@ -51,15 +51,15 @@ Example configuration with distributed or parallel training:
...
+To use distributed training, ``slots_per_trial`` must be set to a multiple
 of the number of GPUs per machine. For example, if you have a cluster of machines with
 8 GPUs each, you should set ``slots_per_trial`` to a multiple of 8, such as 8, 16, or 24.
 This ensures that the full network and interconnect bandwidth are available to
 multi-GPU workloads and results in better performance.
 
 .. warning::
 
-  Distributed and parallel training are designed to maximize performance by
+  Distributed training are designed to maximize performance by
```
Comment (aaron276h) [inline docs/how-to/distributed-training.txt:62]: `Distributed training is ...`

File: docs/how-to/distributed-training.txt [hunk 3]
```diff
@@ -12,19 +12,19 @@ Determined provides three main methods to take advantage of multiple GPUs:
    :ref:`hyperparameter search <topic-guides_hp-tuning-det>` may train more than
    one trial at once, each of which will use its own GPUs.
 3. **Parallelism within a trial.** Use multiple GPUs to speed up the
-   training of a single trial. Determined can coordinate across multiple GPUs on a single machine (parallel training)
-   or across multiple GPUs on multiple machines (distributed training) to improve the
+   training of a single trial. Determined can coordinate across multiple GPUs on a single machine
+   or across multiple GPUs on multiple machines (*distributed training*) to improve the
```
[the remark this answers is not in the document]
Comment (neilconway) [reply docs/how-to/distributed-training.txt:16]: Right -- I meant to say that both clauses of the "or" are included under "distributed training". Reworded.

File: docs/topic-guides/optimizing-distributed-training.txt [hunk 4]
```diff
@@ -1,24 +1,24 @@
 .. _optimizing-multi-gpu-training:
 
-Optimizing Distributed and Parallel Training
+Optimizing Distributed Training
 ============================================
```
Comment (aaron276h) [inline docs/topic-guides/optimizing-distributed-training.txt:4]: nit: shorten underscore to match title

File: harness/determined/pytorch/_callback.py [hunk 5]
```diff
@@ -64,8 +63,8 @@ def on_checkpoint_end(self, checkpoint_dir: pathlib.Path) -> None:
         Run after every checkpoint.
 
         .. warning::
-            This callback executes only on the chief GPU which performs the
-            the checkpoint in the distributed and parallel training settings.
+            This callback currently only executes on the chief GPU when doing
```
Comment (aaron276h) [inline harness/determined/pytorch/_callback.py:66]: I would drop currently here, since we have no plans on changing this.

**Pull request merged**